src/app/user-flight-history/user-flight-history.component.ts

```diff
@@ -25,8 +26,8 @@ export class UserFlightHistoryComponent implements OnInit {
     ) { }
 
   getHistory(): void {
-    const customerId = this.route.snapshot.paramMap.get('id');
-    if (customerId !== null) {
+    const customerId = this.authService.userId;
+    if (customerId !== null && customerId !== undefined) {
```

Comment:
this can be simplified with a JS trick
```
if (!!customerId)
```
That'll make sure it's neither null nor undefined.